coin-flip/app/components/InitPlayerButton.tsx

```diff
@@ -0,0 +1,120 @@
+
+    setIsLoading(true)
+
+    try {
+      // Generate keypair for a new VRF account
+      const vrfAccountKeypair = Keypair.generate()
+
+      // Create a callback instruction to store on the VRF account (the consumeRandomness instruction)
+      const vrfCallbackInstruction: sbv2.Callback = {
+        programId: program.programId,
+        accounts: [
+          { pubkey: publicKey, isSigner: false, isWritable: true },
+          { pubkey: solVaultPDA, isSigner: false, isWritable: true },
+          { pubkey: gameStatePDA, isSigner: false, isWritable: true },
+          {
+            pubkey: vrfAccountKeypair.publicKey,
+            isSigner: false,
+            isWritable: false,
+          },
+          {
+            pubkey: SystemProgram.programId,
+            isSigner: false,
+            isWritable: false,
+          },
+        ],
+        ixData: anchorProgramInstructionCoder.encode("consumeRandomness", ""),
+      }
+
+      // Intructions to create the VRF account
+      const [VrfAccount, TransactionObject] =
+        await sbv2.VrfAccount.createInstructions(
+          switchboard?.program, // switchboard program
+          publicKey, // payer
+          {
+            vrfKeypair: vrfAccountKeypair, // keypair to initialize the new VRF account with
+            queueAccount: switchboard.queueAccount, // switchboard queue account
+            callback: vrfCallbackInstruction, // callback instruction
+            authority: gameStatePDA, // authority of the VRF account (game state PDA)
+          }
+        )
+
+      // Instructions to create the VRF permission account
+      const [PermissionAccount, TransactionObject2] =
+        sbv2.PermissionAccount.createInstruction(
+          switchboard.program, // switchboard program
+          publicKey, // payer
+          {
+            granter: switchboard.queueAccount.publicKey, // the queue account is the granter granting permission to the VRF account
+            grantee: vrfAccountKeypair.publicKey, // the VRF account is the grantee being granted permission to access the queue
+            authority: switchboard.queueAccountData.authority, //  the queue authority is responsible for vrf accounts access to the queue
+          }
+        )
+
+      // Instruction to initialize the player's game state account
+      const instruction = await program.methods
+        .initialize()
+        .accounts({
+          player: publicKey,
+          gameState: gameStatePDA,
+          vrf: VrfAccount.publicKey,
+        })
+        .instruction()
+
+      // Create a transaction with all the instructions
+      const tx = new Transaction().add(
+        ...TransactionObject.ixns,
+        ...TransactionObject2.ixns,
+        instruction
+      )
+
+      // Send the transaction, signing with the VRF account keypair (for initialization of the VRF account)
+      const txSig = await sendTransaction(tx, connection, {
+        signers: [vrfAccountKeypair],
```

Comment:
The instruction to create the VrfAccount is here (there's multiple instructions). The connected wallet (publicKey) is set as payer for the instructions. The keypair is "signing" because its being used to as the address for the VrfAccount that's being initialized (paid for by the connected wallet).
```
const [VrfAccount, TransactionObject] =
  await sbv2.VrfAccount.createInstructions(
    switchboard?.program, // switchboard program
    publicKey, // <--- payer  set here
    {
      vrfKeypair: vrfAccountKeypair, // keypair to initialize the new VRF account with
      queueAccount: switchboard.queueAccount, // switchboard queue account
      callback: vrfCallbackInstruction, // callback instruction
      authority: gameStatePDA, // authority of the VRF account (game state PDA)
    }
  )        
```